File: 0932-monotonic-array/0932-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        increasing = False
        decreasing = False
        l = len(nums)
        for i in range(1, l):
            if nums[i] > nums[i-1]:
                increasing = True
                if increasing == decreasing:
                    return False
            elif nums[i] < nums [i-1]:
                decreasing = True
                if increasing == decreasing:
                    return False
        return True                  

File: 0932-monotonic-array/0932-monotonic-array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function(nums) {
        let increasing = false;
        let decreasing = false;
        const len = nums.length;
        
        for (let i = 1; i < len; i++) {
            if (nums[i] > nums[i - 1]) {
                increasing = true;
                if (increasing === decreasing) {
                    return false;
                }
            } else if (nums[i] < nums[i - 1]) {
                decreasing = true;
                if (increasing === decreasing) {
                    return false;
                }
            }
        }
        return true;
    }
